File: app/src/main/java/com/example/lorddarthart/weathertest/OnItemTouchListener.java
package com.example.lorddarthart.weathertest;

import android.view.View;

public interface OnItemTouchListener {
    /**
     * Callback invoked when the user Taps one of the RecyclerView items
     *
     * @param view     the CardView touched
     * @param position the index of the item touched in the RecyclerView
     */
    void onCardViewTap(View view, int position);

    /**
     * Callback invoked when the Button1 of an item is touched
     *
     * @param view     the Button touched
     * @param position the index of the item touched in the RecyclerView
     */

    void onButtonCvMenuClick(View view, int position);

}

File: app/src/main/java/tk/lorddarthart/weathertest/util/OnItemTouchListener.kt
package tk.lorddarthart.weathertest.util

import android.view.View

interface OnItemTouchListener {
    /**
     * Callback invoked when the user Taps one of the RecyclerView items
     *
     * @param view     the CardView touched
     * @param position the index of the item touched in the RecyclerView
     */
    fun onCardViewTap(view: View, position: Int)

    /**
     * Callback invoked when the Button1 of an item is touched
     *
     * @param view     the Button touched
     * @param position the index of the item touched in the RecyclerView
     */

    fun onButtonCvMenuClick(view: View, position: Int)

}